Update upstream Prometheus

pracucci opened the pull request:
I want to get https://github.com/prometheus/prometheus/pull/11897.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/grafana/mimir-prometheus?pullRequest=406) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/grafana/mimir-prometheus?pullRequest=406) before we can accept your contribution.<br/>**2** out of **7** committers have signed the CLA.<br/><br/>:white_check_mark: beorn7<br/>:white_check_mark: pracucci<br/>:x: roidelapluie<br/>:x: saminzadeh<br/>:x: mmorel-35<br/>:x: kakkoyun<br/>:x: bwplotka<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/grafana/mimir-prometheus?pullRequest=406) it.</sub>

pstibrany reviewed (approved):
I am surprised to see so many changes since last upstream merge yesterday.

Please let's not forget to merge with merge commit.

pracucci commented:
> I am surprised to see so many changes since last upstream merge yesterday.

Me too. I checked commits in Prometheus and apparently they were merged.

pstibrany commented:
Ah, I see. It's lot of commits coming from just few not-squashed merges.
